bundle/file.go

```diff
@@ -265,3 +269,93 @@ func (t *tarballLoader) NextFile() (*Descriptor, error) {
 
 	return newDescriptor(path.Join(t.baseURL, f.name), f.name, f.reader), nil
 }
+
+// Next implements the storage.Iterator interface.
+// It iterates to the next policy or data file in the directory tree
+// and returns a storage.Update for the file.
+func (t *tarballLoader) Next() (*storage.Update, error) {
+	if t.tr == nil {
+		gr, err := gzip.NewReader(t.r)
+		if err != nil {
+			return nil, errors.Wrap(err, "archive read failed")
+		}
+
+		t.tr = tar.NewReader(gr)
+	}
+
+	if t.files == nil {
+		t.files = []file{}
+
+		for {
+			header, err := t.tr.Next()
+			if err == io.EOF {
+				break
+			}
+
+			if err != nil {
+				return nil, err
+			}
+
+			// Keep iterating on the archive until we find a normal file
+			if header.Typeflag == tar.TypeReg {
+
+				if filepath.Base(header.Name) == dataFile || filepath.Base(header.Name) == yamlDataFile || strings.HasSuffix(header.Name, RegoExt) {
+					f := file{name: header.Name}
+
+					var buf bytes.Buffer
+					if _, err := io.Copy(&buf, t.tr); err != nil {
+						return nil, errors.Wrapf(err, "failed to copy file %s", header.Name)
+					}
+
+					f.reader = &buf
+
+					t.files = append(t.files, f)
+				}
+			}
+		}
+		sortFilePathAscend(t.files)
```

Comment:
I'm assuming this sort is done to ensure that files in the bundle are written into the store in a deterministic order. It would be good to include a test case that asserts on this behaviour.

Reply:
Added a test case.